File: aat-lib/src/main/java/ch/bailu/aat_lib/view/graph/GraphPainterLimit.java
package ch.bailu.aat_lib.view.graph;

import ch.bailu.aat_lib.gpx.GpxPointNode;

public class GraphPainterLimit extends GraphPainter {
    private int index = 0;
    private final Segment segment;

    public GraphPainterLimit(GraphPlotter p, Segment segment, int md) {
        super(p, md);
        this.segment = segment;
    }

    @Override
    public void doPoint(GpxPointNode point) {
        if (segment.isInside(index)) {
            super.doPoint(point);
        }
        index++;
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/view/graph/GraphPainterLimit.kt
package ch.bailu.aat_lib.view.graph

import ch.bailu.aat_lib.gpx.GpxPointNode

class GraphPainterLimit(p: GraphPlotter, private val segment: Segment, md: Int) :
    GraphPainter(p, md) {
    private var index = 0

    override fun doPoint(point: GpxPointNode) {
        if (segment.isInside(index)) {
            super.doPoint(point)
        }
        index++
    }
}
